libraries/chain/asset_evaluator.cpp

```diff
@@ -466,6 +411,7 @@ void_result asset_update_bitasset_evaluator::do_evaluate(const asset_update_bita
 
    FC_ASSERT( !current_bitasset_data.has_settlement(), "Cannot update a bitasset after a global settlement has executed" );
 
+   // hf 922_931 is a consensus/logic change. This hf cannot be removed.
```

Comment:
Which logic change below is triggered by which asset(s) at which block(s)? Mainnet or testnet?

Reply:
I hope this answers your question. I added an "else" to this "if" and logged if the test failed. At 16% replayed on mainnet, I grep'd the following from the logs:

```2019-09-05T19:07:46        th_a:?unnamed?     cond_wlog_assert ] Before HF 922/931: Cannot update an asset to be backed by itself. at block 20231047			asset_evaluator.cpp:404
2019-09-05T19:18:25        th_a:?unnamed?     cond_wlog_assert ] Before HF 922/931: Cannot update an asset to be backed by itself. at block 23721657			asset_evaluator.cpp:404
2019-09-05T19:20:33        th_a:?unnamed?     cond_wlog_assert ] Before HF 922/931: Cannot update an asset to be backed by itself. at block 24080299			asset_evaluator.cpp:404
2019-09-05T19:21:48        th_a:?unnamed?     cond_wlog_assert ] Before HF 922/931: Cannot update an asset to be backed by itself. at block 24276618			asset_evaluator.cpp:404
2019-09-05T19:21:48        th_a:?unnamed?     cond_wlog_assert ] Before HF 922/931: A BitAsset cannot be backed by a BitAsset that itself is backed by a BitAsset. at block 24276618			asset_evaluator.cpp:404
2019-09-05T19:22:05        th_a:?unnamed?     cond_wlog_assert ] Before HF 922/931: Cannot update an asset to be backed by itself. at block 24328794			asset_evaluator.cpp:404
```

I believe this shows that by removing the hardfork code, consensus would not be maintained. Let me know if you need more information.

Reply:
Thanks.